### PR TITLE
Implement wrapper in `fusion-test-utils` to encapsulate plugin resolution

### PR DESCRIPTION
**Pull request by AlexMSmithCA**
Fixes #364

Primary changes:
* Add `getSimulator` to encapsulate plugin resolution and expose `render` and `request`
* Add Flow types, where appropriate
* Update unit tests - includes addition of test with resolution of plugin dependencies
* Update `README.md`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,32 +1,65 @@
...
-export function render(app, url, options = {}) {
+const render = (app: FusionApp) => (
+  url: string,
+  options: * = {}
+): Promise<*> => {
   const ctx = renderContext(url, options);
   return simulate(app, ctx);
+};
+
+export function registerAsTest(
```
Comment (AlexMSmithCA) [inline src/index.js]: Open to thoughts here on the exact API and naming this function.  Primary motivation here is to encapsulate resolving the app to allow consumers to have a single app fixture and call `.request` or `.render` multiple times (which you cannot do today).

File: yarn.lock [hunk 2]
```diff
@@ -2802,6 +2802,10 @@ fusion-core@0.3.0-2:
     node-mocks-http "^1.6.6"
     toposort "^1.0.6"
 
+fusion-tokens@^0.0.4:
+  version "0.0.4"
+  resolved "https://unpm.uberinternal.com/fusion-tokens/-/fusion-tokens-0.0.4.tgz#b84c58e2de8e06d3e63c2c182da7e023ccfb50ec"
```
Comment (KevinGrandon) [inline yarn.lock]: 💣

File: src/index.js (hunk 1 above)
Comment (lhorie) [reply src/index.js]: `createSimulator`? `createTestDriver`?

File: src/index.js [hunk 3]
```diff
@@ -1,32 +1,62 @@
...
+  url: string,
+  options: * = {}
+): Promise<*> => {
   const ctx = renderContext(url, options);
   return simulate(app, ctx);
+};
+
+export function getSimulator(app: FusionApp, testPlugin?: FusionPlugin<*, *>) {
+  if (testPlugin) {
+    app.register(testPlugin);
```
Comment (KevinGrandon) [inline src/index.js]: Should we allow for an array of test plugins?

File: .flowconfig [hunk 4]
```diff
@@ -1,14 +1,15 @@
 [ignore]
 .*/node_modules/.*[^(package)]\.json$
+<PROJECT_ROOT>/dist/.*
 
 [include]
 ./src/
 
 [libs]
+./node_modules/fusion-core/flow-typed
```
Comment (KevinGrandon) [inline .flowconfig]: Are we using this for anything yet? Or just getting our configs to match?

File: src/index.js (hunk 3 above)
Comment (lhorie) [reply src/index.js]: This is just the public API from the DI system. Whether multiple registration makes sense is something that we can discuss on Giancarlo's new proposal.

File: .flowconfig (hunk 4 above)
Comment (AlexMSmithCA) [reply .flowconfig]: Good question- it's used here for `[Context](https://github.com/fusionjs/fusion-test-utils/pull/38/commits/8ba7a6b51553d4e6b0eb380479e853d063339882#diff-9264294e9bb5e1469869da14a9cea86dR5)` and `FusionApp`.  At least internally, it looks like our plan is to ship `libdef` files like we are will soon be doing in `[fusion-core](https://github.com/fusionjs/fusion-core/blob/di/flow-typed/fusion-core.js)`.

**Pull request merged**